pkg/sdkserver/sdkserver.go

```diff
@@ -275,21 +273,41 @@ func (s *SDKServer) updateState() error {
 
 	s.gsUpdateMutex.RLock()
 	gs.Status.State = s.gsState
+
+	// if we are setting the Reserved status, check for the duration, and set that too
+	if gs.Status.State == stablev1alpha1.GameServerStateReserved && s.gsReserveDuration != nil {
+		n := metav1.NewTime(time.Now().Add(*s.gsReserveDuration))
+		gs.Status.ReservedUntil = &n
+	} else {
+		gs.Status.ReservedUntil = nil
+	}
 	s.gsUpdateMutex.RUnlock()
 
 	_, err = gameServers.Update(gs)
 	if err != nil {
 		return errors.Wrapf(err, "could not update GameServer %s/%s to state %s", s.namespace, s.gameServerName, gs.Status.State)
 	}
 
+	message := "SDK state change"
 	level := corev1.EventTypeNormal
 	// post state specific work here
 	switch gs.Status.State {
 	case stablev1alpha1.GameServerStateUnhealthy:
 		level = corev1.EventTypeWarning
+	case stablev1alpha1.GameServerStateReserved:
+		s.gsUpdateMutex.RLock()
+		hasDuration := s.gsReserveDuration != nil
+		if hasDuration {
+			message += fmt.Sprintf(", for %v", s.gsReserveDuration.String())
```

Comment:
Does this output nicely? The new state is passed as a separate argument than the message, so it isn't clear to me how this will actually appear in the event output. Also, should the message contain units for the duration (or are those appended automatically)?

Reply:
String returns a string representing the duration in the form "72h3m0.5s".

Reply:
The final message will be 'SDK state change, for 10m'. I also think it lacks of information even though the reason is `Reserved`. I would prefer `SDK state change to Reserved for ...' or even better `SDK state change to Reserved until utc date` since when looking at the event you might not know if it is gone.

Also you can drop the `.String()`, %v use the `Stringer` interface if implemented which is your case.

Reply:
> Also you can drop the .String(), %v use the Stringer interface if implemented which is your case.

Done!

> The final message will be SDK state change, for 10m. I also think it lacks of information even though the reason is Reserved. I would prefer SDK state change to Reserved for ... or even better SDK state change to Reserved until utc date since when looking at the event you might not know if it is gone.

To give an example of the full event output, that might help with context on this:

```
Events:
  Type    Reason          Age               From                   Message
  ----    ------          ----              ----                   -------
  Normal  PortAllocation  53s               gameserver-controller  Port allocated
  Normal  Creating        53s               gameserver-controller  Pod simple-udp-567rr created
  Normal  Scheduled       53s               gameserver-controller  Address and port populated
  Normal  RequestReady    47s               gameserver-sidecar     SDK state change
  Normal  Reserved        10s               gameserver-sidecar     SDK state change, for 10s
  Normal  Ready           0s (x2 over 47s)  gameserver-controller  SDK.Ready() complete
```

So you can see the State in the `Reason` column, so I figure it's redundant to include it again in the Message.

I would also leave the time period as `for 10s` - reason being, from the `Age` column, you can actually see how long its been since the `Reserved` state was set, and therefore know what the difference is going to be, and can calculate the time left.

How do you both feel about this now?